### PR TITLE
fix(auth): check the key allowlist on team alias targets

### PR DESCRIPTION
**Pull request by songkuan-zheng**
## TLDR

Problem this solves:

- A team model alias satisfied the model access check on its own
- A key scoped to some of the team's models could call every aliased model
- The bypass applied to the key check and the team check alike

How it solves it:

- The alias is resolved to its target before the allowlist check
- The target goes through the same allowlist, wildcard and access group rules as a plain name
- Unrestricted keys and the all-team-models sentinel behave as before

## User Flow

Before: a developer holding a key scoped to one model can call a more expensive team model by using the team's alias for it

1. The proxy admin sends POST https://litellm-domain/team/new with `"models": ["claude-haiku", "claude-sonnet"]` and `"model_aliases": {"team-fast": "claude-haiku"}`
2. The admin sends POST https://litellm-domain/key/generate for that team with `"models": ["claude-sonnet"]` and hands the key to the developer
3. The developer sends POST https://litellm-domain/v1/chat/completions with `"model": "team-fast"` using that key
4. The response is HTTP 200 with a completion from claude-haiku, a model the key was never granted, and the spend lands on that key
5. Any key on the team can do the same for every alias the team defines, whatever the key's own model list says

After: the same request is refused, and the key still works for the models it was granted

1. The proxy admin sends POST https://litellm-domain/team/new with `"models": ["claude-haiku", "claude-sonnet"]` and `"model_aliases": {"team-fast": "claude-haiku"}`
2. The admin sends POST https://litellm-domain/key/generate for that team with `"models": ["claude-sonnet"]` and hands the key to the developer
3. The developer sends POST https://litellm-domain/v1/chat/completions with `"model": "team-fast"` using that key
4. The response is HTTP 403 with `key not allowed to access model. This key can only access models=['claude-sonnet']. Tried to access team-fast`
5. The same key sending `"model": "claude-sonnet"` still gets HTTP 200, and a sibling key granted `claude-haiku` still gets HTTP 200 through `team-fast`
6. A key can no longer reach a team model through its alias unless the alias target is on the key's own model list

## Relevant issues

No existing issue. The team model alias lookup in the model access check returned an allow as soon as the requested name was an alias key, without ever comparing the alias target against the calling entity's model list. A key created with a narrower `models` list than its team therefore inherited access to every aliased team model, and a team whose own `models` list did not contain an alias target was still allowed through that alias

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have added meaningful tests
- [x] The handful of test files covering my change pass locally, e.g. `uv run pytest tests/test_litellm/<your_test_file>.py -v`. Leave the suites (`make test-unit-*`, `make test-unit`) to CI: it finishes in ~15 minutes where a laptop takes an hour or more
- [x] My PR passes all required CI/CD checks (e.g., lint, schema.d.ts sync check, etc.)
- [x] My PR's scope is as isolated as possible; it only solves 1 specific problem
- [ ] I have received a Greptile **Confidence Score of at least 4/5** before requesting a maintainer review (Greptile reviews automatically once the PR is opened; only comment `@greptileai` to re-request a review after pushing changes)

## Delays in PR merge?

If you're seeing a delay in your PR being merged, ping the LiteLLM Team on [Slack (#pr-review)](https://join.slack.com/t/litellmossslack/shared_invite/zt-3o7nkuyfr-p_kbNJj8taRfXGgQI1~YyA).

## Screenshots / Proof of Fix

Setup shared by both runs. Proxy started from the worktree with `uv run python litellm/proxy/proxy_cli.py --config config.yaml --port 4101` against a fresh `postgres:16` container. The two deployments hit a real Anthropic-compatible gateway, so every 200 below cost real tokens

```yaml
model_list:
  - model_name: claude-haiku
    litellm_params:
      model: anthropic/as-anthropic/claude-haiku-4-5-20251001
      api_key: os.environ/ANTHROPIC_API_KEY
      api_base: os.environ/ANTHROPIC_API_BASE
  - model_name: claude-sonnet
    litellm_params:
      model: anthropic/as-anthropic/claude-sonnet-5
      api_key: os.environ/ANTHROPIC_API_KEY
      api_base: os.environ/ANTHROPIC_API_BASE
general_settings:
  master_key: sk-up-team-alias-master
  database_url: postgresql://postgres:uppw@localhost:5441/litellm
```

Team and keys are created the same way on both sides. `$M` is the master key header, `$TEAM` is the id returned by `/team/new`, and the chat payload is always `{"model": ..., "messages": [{"role": "user", "content": "Reply with the single word ok"}], "max_tokens": 5}`

```bash
curl -s http://localhost:4101/team/new -H "$M" -H 'Content-Type: application/json' \
  -d '{"team_alias":"alias-demo","models":["claude-haiku","claude-sonnet"],"model_aliases":{"team-fast":"claude-haiku"}}'
curl -s http://localhost:4101/key/generate -H "$M" -H 'Content-Type: application/json' \
  -d "{\"team_id\":\"$TEAM\",\"models\":[\"claude-sonnet\"],\"key_alias\":\"sonnet-only\"}"
curl -s http://localhost:4101/key/generate -H "$M" -H 'Content-Type: application/json' \
  -d "{\"team_id\":\"$TEAM\",\"models\":[\"claude-haiku\"],\"key_alias\":\"haiku\"}"
```

#### Key scoped to claude-sonnet calls the team alias team-fast

1. `curl -s -w '\nHTTP %{http_code}\n' http://localhost:4101/v1/chat/completions -H "Authorization: Bearer $SONNET_ONLY_KEY" -H 'Content-Type: application/json' -d '{"model":"team-fast", ...}'`
2. Observed: `{"model": "team-fast", "content": "ok", "usage": 17}` and `HTTP 200`. The key was granted `claude-sonnet` only, yet the alias took it to `claude-haiku`

#### Key granted claude-haiku calls the team alias team-fast

1. `curl -s -w '\nHTTP %{http_code}\n' http://localhost:4101/v1/chat/completions -H "Authorization: Bearer $HAIKU_KEY" -H 'Content-Type: application/json' -d '{"model":"team-fast", ...}'`
2. Observed: `{"model": "team-fast", "content": "ok", "usage": 17}` and `HTTP 200`

#### Key scoped to claude-sonnet calls the team alias team-fast

1. `curl -s -w '\nHTTP %{http_code}\n' http://localhost:4101/v1/chat/completions -H "Authorization: Bearer $SONNET_ONLY_KEY" -H 'Content-Type: application/json' -d '{"model":"team-fast", ...}'`
2. Observed: `{"error": {"message": "key not allowed to access model. This key can only access models=['claude-sonnet']. Tried to access team-fast", "type": "key_model_access_denied", "param": "model", "code": "403"}}` and `HTTP 403`

#### Key scoped to claude-sonnet calls its own model

1. `curl -s -w '\nHTTP %{http_code}\n' http://localhost:4101/v1/chat/completions -H "Authorization: Bearer $SONNET_ONLY_KEY" -H 'Content-Type: application/json' -d '{"model":"claude-sonnet", ...}'`
2. Observed: `{"model": "claude-sonnet", "content": "ok", "usage": 19}` and `HTTP 200`

#### Key granted claude-haiku calls the team alias team-fast

1. `curl -s -w '\nHTTP %{http_code}\n' http://localhost:4101/v1/chat/completions -H "Authorization: Bearer $HAIKU_KEY" -H 'Content-Type: application/json' -d '{"model":"team-fast", ...}'`
2. Observed: `{"model": "team-fast", "content": "ok", "usage": 17}` and `HTTP 200`

## Type

🐛 Bug Fix

### Severe

- Keys that relied on an alias to reach a model outside their own list now get 403
- Teams whose `models` list omits an alias target now get 403 through that alias

### Low

- The header forwarding lookup by model group shares the helper and now resolves aliases to their target too
- A team alias name is no longer matched against the allowlist itself, and it is not resolved through the router or global alias maps; only its target is, so a name that is both a team alias and a router alias follows the team alias, as request routing already does

## Final Attestation

- [x] The tests check the right things, including the edge cases, and regressions in the respective real-world customer use-cases are not possible after this PR

### CODE REVIEW

File: litellm/proxy/auth/auth_checks.py [hunk 1]
```diff
@@ -3959,7 +3959,13 @@ def _check_model_access_helper(
...
     filtered_models: Final = [m for m in models if m not in access_groups]
 
-    if _model_in_team_aliases(model=model, team_model_aliases=team_model_aliases):
+    team_alias_target: Final = team_model_aliases.get(model) if team_model_aliases else None
+    if team_alias_target is not None and _check_model_access_helper(
+        model=team_alias_target,
+        llm_router=llm_router,
+        models=models,
+        team_id=team_id,
+    ):
```
[the remark this answers is not in the document]
Comment (songkuan-zheng) [reply litellm/proxy/auth/auth_checks.py]: Addressed in 83f52d8985. The helper now resolves the team alias to its target before any matching and runs the wildcard, exact and access-group checks on the target only, so the alias name is never matched against the allowlist. Covered by test_can_key_call_model_team_alias_name_in_key_models_is_not_enough, which denies a key whose list holds the alias name or a wildcard matching it

[the remark this answers is not in the document]
Comment (songkuan-zheng) [reply litellm/proxy/auth/auth_checks.py]: Addressed in 83f52d8985. _can_object_call_model resolves the team alias first and then applies the existing router and global alias expansion to the resolved target, so a team alias that points at a router model-group alias is allowed when the key lists the underlying model. Covered by test_can_object_call_model_team_alias_target_expands_router_alias with a real Router carrying model_group_alias

File: litellm/proxy/auth/auth_checks.py [hunk 2]
```diff
@@ -3959,7 +3959,13 @@ def _check_model_access_helper(
     # Filter out models that are access_groups
     filtered_models: Final = [m for m in models if m not in access_groups]
 
-    if _model_in_team_aliases(model=model, team_model_aliases=team_model_aliases):
+    team_alias_target: Final = team_model_aliases.get(model) if team_model_aliases else None
```
[the remark this answers is not in the document]
Comment (songkuan-zheng) [reply litellm/proxy/auth/auth_checks.py]: Addressed in 83f52d8985. When a name is a team alias, only the team target is considered; the alias name is not expanded through litellm.model_alias_map or the router model_group_alias any more, which matches the precedence request routing applies. Covered by test_can_object_call_model_team_alias_takes_precedence_over_router_alias

**Comment by songkuan-zheng** [comment]
@greptileai

**Comment by songkuan-zheng** [comment]
Ready for maintainer review, Greptile 5/5 on 83f52d8985
